config/kernel-generic_readlink.m4

```diff
@@ -0,0 +1,23 @@
+dnl #
+dnl # 4.10 API
+dnl #
+dnl # NULL inode_operations.readlink implies generic_readlink(), which
+dnl # has been made static.
+dnl #
+AC_DEFUN([ZFS_AC_KERNEL_GENERIC_READLINK_GLOBAL], [
+	AC_MSG_CHECKING([whether generic_readlink is global])
+	ZFS_LINUX_TRY_COMPILE([
+		#include <linux/fs.h>
+
+		extern int i;
+
+		i = generic_readlink(NULL, NULL, 0);
```

Comment:
This check should be done in the function body and not global.  I'd expect this to always fail regardless of if `generic_readlink()` is available.